include/solo5/mft_abi.h

```diff
@@ -99,55 +100,76 @@ struct mft {
 };
 
 /*
- * Manifest ELF note headers.
+ * Maximum supported number of manifest entires.
+ */
+#define MFT_MAX_ENTRIES 64
+
+/*
+ * HERE BE DRAGONS.
  *
- * Note that somewhat counter-intuitively, in practice the note section header
- * fields (namesz/descsz/type) are 32-bit words even on ELF64.
+ * The following structures and macros are used to declare a Solo5 "MFT1"
+ * format note at link time. This is somewhat tricky, as we need to ensure all
+ * structures are aligned correctly.
  */
-#define SOLO5_NOTE_NAME     "Solo5"
-#define SOLO5_NOTE_MANIFEST 0x3154464d /* "MFT1" */
+#define SOLO5_NOTE_NAME      "Solo5"
+#define SOLO5_NOTE_TYPE_MFT1 0x3154464d /* "MFT1" */
 
-struct mft_note_header {
-    uint32_t namesz;
-    uint32_t descsz;
-    uint32_t type;
-    char name[(sizeof(SOLO5_NOTE_NAME) + 3) & -4];
+/*
+ * Defines an Elf64_Nhdr with n_name filled in and padded to a 4-byte boundary,
+ * i.e. the common part of a Solo5-owned Nhdr.
+ */
+struct mft1_nhdr {
+    uint32_t n_namesz;
+    uint32_t n_descsz;
+    uint32_t n_type;
+    char n_name[(sizeof(SOLO5_NOTE_NAME) + 3) & -4];
 };
```

Comment:
Since this is an on-disk wire format, would it make sense to have something here to guarantee that padding etc makes sense?
Self-alignment for the 32bit members makes sense, but it seems to me that whether or not padding is applied to the `struct mft1_nhdr` would depend on `SOLO5_NOTE_NAME`.
I'm thinking either adding `__attribute__((packed))` or
```c
_Static_assert((sizeof(struct mft1_ndr) == (4+4+4+8)), "mft1_nhdr alignment issue");
```
(or both?)

Same re: `mft1_note` below, ensuring that
```c
_Static_assert(sizeof(struct mft1_note) == sizeof(struct mft1_nhdr) + sizeof(struct mft),
        "struct mft1_note would be padded");
```
(because we have occurrences of `sizeof(struct mft1_note) - sizeof(struct mft1_nhdr)` which rests on the variant that there's no padding between the two)

Reply:
Thanks for the suggestions. I've merged #388 which switches compilation to `-std=c11`, enabling us to use `_Static_assert()`.

The first `_Static_assert()` makes sense, though I would do:

```
_Static_assert((sizeof(struct mft1_nhdr)) ==
        (sizeof(uint32_t) + sizeof(uint32_t) + sizeof(uint32_t) + 8),
        "struct mft1_nhdr alignment issue");
```

The 2nd one will not pass, try it yourself. Dealing with the padding between `h` and `m` in a `struct mft1_note` is precisely why this change introduces `note_align`, `MFT1_NOTE_ALIGN` and related. Thus, `sizeof(struct mft1_note) - sizeof(struct mft1_nhdr)` must include the padding.

Reply:
Ah, okay, my bad!